comparative_benchmark/tf_xla/run_benchmarks.py

```diff
@@ -0,0 +1,163 @@
+        tf.test.experimental.sync_devices()
+        end = time.perf_counter()
+        warmup_latencies.append(1000 * (end - start))
+
+      # Run benchmark.
+      latencies = []
+      last_outputs = None
+      for i in range(benchmark_iterations):
+        start = time.perf_counter()
+        last_outputs = model_obj.forward(inputs)
+        tf.test.experimental.sync_devices()
+        end = time.perf_counter()
+        latencies.append(1000 * (end - start))
+
+      if last_outputs is None:
+        raise ValueError("No benchmark runs.")
+
+      verdicts = utils.compare_tensors(outputs=last_outputs,
+                                       expects=expects,
+                                       **verify_params)
+      all_equal = True
+      for idx, verdict in enumerate(verdicts):
+        is_equal, max_diff = verdict
+        if not is_equal:
+          print(f"Output {idx} exceeds tolerance. Max diff: {max_diff}")
+          all_equal = False
+
+      if not all_equal:
+        raise ValueError("Output verification failed.")
+
+      # Retrieve memory stats.
+      if tf_device == _TF_GPU_DEVICE:
+        memory_info = tf.config.experimental.get_memory_info(tf_device)
+        device_peak_b = memory_info["peak"]
+      else:
+        # tf.config.experimental does not currently support measuring CPU memory usage.
+        device_peak_b = None
+      device_peak_mb = bytes_to_mb(device_peak_b)
+
+      # Roughly calculate compile time.
+      compile_time_s = None if not warmup_latencies else (
+          max(warmup_latencies) - statistics.median(latencies)) / 1000
+
+  except Exception as e:
+    print(f"Failed to benchmark model {model.name}. Exception: {e}")
+    raise RuntimeError(e)
+
+  # Save results.
+  result_dict = {
+      "min_warmup_latency_ms":
+          min(warmup_latencies, default=None),
+      "max_warmup_latency_ms":
+          max(warmup_latencies, default=None),
+      "mean_warmup_latency_ms":
+          None if not warmup_latencies else statistics.mean(warmup_latencies),
+      "median_warmup_latency_ms":
+          None if not warmup_latencies else statistics.median(warmup_latencies),
+      "stddev_warmup_latency_ms":
+          None if not warmup_latencies else statistics.stdev(warmup_latencies),
+      "warmup_iterations":
+          warmup_iterations,
+      "min_latency_ms":
+          min(latencies, default=None),
+      "max_latency_ms":
+          max(latencies, default=None),
+      "mean_latency_ms":
+          None if not latencies else statistics.mean(latencies),
+      "median_latency_ms":
+          None if not latencies else statistics.median(latencies),
+      "stddev_latency_ms":
+          None if not latencies else statistics.stdev(latencies),
+      "benchmark_iterations":
+          benchmark_iterations,
+      "compile_time_s":
+          compile_time_s,
+      "device_memory_peak_mb":
+          device_peak_mb,
+  }
+  shared_dict.update(result_dict)
```

Comment:
Feel like we don't need to update `shared_dict` here. Instead we can return the `result_dict` and let `benchmark_lib` update the dict? (So one fewer parameter to pass to this function)

Reply:
Yeah that makes sense. Concurrency details should be hidden in `benchmark_lib`. Updated.